AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_LightAdded.py

```diff
@@ -97,74 +182,299 @@ def AtomEditorComponents_Light_AddedToEntity():
 
-        # 4. REDO the entity creation and component addition.
-        # -> REDO entity creation.
-        general.redo()
-        # -> REDO selecting entity.
-        general.redo()
-        # -> REDO naming entity.
-        general.redo()
-        # -> REDO component addition.
-        general.redo()
-        general.idle_wait_frames(1)
-        Report.result(Tests.creation_redo, light_entity.exists())
+            # 3. Add Light component to the Light entity.
+            light_component = light_entity.add_component(AtomComponentProperties.light())
+            if iterations == 0:
+                Report.critical_result(Tests.light_component,
+                                       light_entity.has_component(AtomComponentProperties.light()))
 
-        # 5. Cycle through all light types.
-        for light_type in LIGHT_TYPES.keys():
+                # 4. UNDO the entity creation and component addition.
+                # -> UNDO component addition.
+                general.undo()
+                # -> UNDO naming entity.
+                general.undo()
+                # -> UNDO selecting entity.
+                general.undo()
+                # -> UNDO entity creation.
+                general.undo()
+                general.idle_wait_frames(1)
+                Report.result(Tests.creation_undo, not light_entity.exists())
+
+                # 5. REDO the entity creation and component addition.
+                # -> REDO entity creation.
+                general.redo()
+                # -> REDO selecting entity.
+                general.redo()
+                # -> REDO naming entity.
+                general.redo()
+                # -> REDO component addition.
+                general.redo()
+                general.idle_wait_frames(1)
+                Report.result(Tests.creation_redo, light_entity.exists())
+
+                # Update loop iterations to prevent additional running of these tests
+                iterations += 1
+
+            # 6. Set light type.
             light_component.set_component_property_value(
                 AtomComponentProperties.light('Light type'), LIGHT_TYPES[light_type])
             current_light_type = light_component.get_component_property_value(
                 AtomComponentProperties.light('Light type'))
             test_light_type = (
-                f"Light component has {light_type} type set",
-                f"Light component failed to set {light_type} type")
+                f"Set light type: {light_type.upper()}",
+                f"P0: Light component failed to set {light_type} type")
+            general.idle_wait_frames(1)
             Report.result(test_light_type, current_light_type == LIGHT_TYPES[light_type])
 
-        # 6. Enter/Exit game mode.
-        TestHelper.enter_game_mode(Tests.enter_game_mode)
-        general.idle_wait_frames(1)
-        TestHelper.exit_game_mode(Tests.exit_game_mode)
+            # Exit loop for unknown light type.
```

Comment:
this can be removed when `LIGHT_TYPES.pop('unknown')` is added before iterating the for loop

Reply:
I don't actually want to remove the 'unknown' type. It is a valid selection showing as "Choose a light type" and has some basic functions of light components such as removing existing shape components from the entity.

Reply:
I kept 'unknown' for now, in the event it is needed at some future time - rather than doing the LIGHT_TYPES.pop('unknown'). Let me know if you are strongly opposed to this.

I did however move it to the top of the loop so it's just ignored immediately.

It may have use in the future if we decide to do any additional testing involving setting the light type back to 'unknown' as this is a valid selection by the user.

Reply:
we can use copy to get around that:
```
light_types_copy = LIGHT_TYPES.copy()
        light_types_copy.pop('unknown')
        # Cycle through light types to test component properties.
        for light_type in light_types_copy.keys():
```

Reply:
Updated